plugins/node/instrumentation-fs/src/constants.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { FMember, FPMember } from './types';
+
+export const PROMISE_FUNCTIONS: FPMember[] = [
+  'access',
+  'appendFile',
+  'chmod',
+  'chown',
+  'copyFile',
+  'cp' as FPMember, // added in v16
+  'lchown',
+  'link',
+  'lstat',
+  'lutimes', // added in v12
+  'mkdir',
+  'mkdtemp',
+  'open',
+  'opendir', // added in v12
+  'readdir',
+  'readFile',
+  'readlink',
+  'realpath',
+  'rename',
+  'rm', // added in v14
+  'rmdir',
+  'stat',
+  'symlink',
+  'truncate',
+  'unlink',
+  'utimes',
+  'writeFile',
+  // 'lchmod', // only implemented on macOS
+];
+
+export const ASYNC_FUNCTIONS: FMember[] = [
+  'access',
+  'appendFile',
+  'chmod',
+  'chown',
+  'copyFile',
+  'cp' as FMember, // added in v16
+  'exists', // deprecated, inconsistent cb signature
+  'lchown',
+  'link',
+  'lstat',
+  'lutimes', // added in v12
+  'mkdir',
+  'mkdtemp',
+  'open',
+  'opendir', // added in v12
+  'readdir',
+  'readFile',
+  'readlink',
+  'realpath',
+  'rename',
+  'rm', // added in v14
+  'rmdir',
+  'stat',
+  'symlink',
+  'truncate',
+  'unlink',
+  'utimes',
+  'writeFile',
+  // 'close', // functions on file descriptor
+  // 'fchmod', // functions on file descriptor
+  // 'fchown', // functions on file descriptor
+  // 'fdatasync', // functions on file descriptor
+  // 'fstat', // functions on file descriptor
+  // 'fsync', // functions on file descriptor
+  // 'ftruncate', // functions on file descriptor
+  // 'futimes', // functions on file descriptor
+  // 'lchmod', // only implemented on macOS
```

Comment:
is there a reason not to implement it nevertheless?

Reply:
No good one.
Just wanted to get the core of it done and included everything that came "free" for the first version of it.

Reply:
Thinks its fine to get something out and iterate

Reply:
👍